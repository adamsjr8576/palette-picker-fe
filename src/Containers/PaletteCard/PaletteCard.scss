.palette-article {
  width: 250px;
  height: 400px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-end;
}
.palette-info-container {
  width: 80%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  height: 100px;
}

.hex-code-section {
  width: 50%;
  min-width: 65px;
  border-radius: 20px;
  background-color: rgba(0, 0, 0, .4);
  padding: 5px;
  margin-bottom: 10px;
  color: white;
}

.lock-button-section {
  background-color: rgba(255, 255, 255, .6);
  border-radius: 50%;
  width: 55px;
  height: 55px;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.lock-button-image {
  height: 35px;
  width: 35px;
}

.hex-p {
  margin: 0;
}
