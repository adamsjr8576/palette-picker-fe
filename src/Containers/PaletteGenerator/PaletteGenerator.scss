.palette-section {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  height: 95vh;
  width: 70%;
}

.palette-card-section {
  display: flex;
  align-items: flex-start;
  height: 87%;
  width: 100%;
}

.palette-button {
  font-size: 2em;
  padding: 10px;
  border-radius: 10px;
  border: 0;
  margin-bottom: 20px;
  outline: none;
  background-color: #444343;
  color: white;
  cursor: pointer;
}

.palette-button:hover {
  box-shadow: 3px 5px #203246;
}

.palette-button:focus {
  border: 2px solid #203246;
}

.palette-button:active {
  transform: translate(3px, 3px);
  box-shadow: 1px 2px #203246;
}
