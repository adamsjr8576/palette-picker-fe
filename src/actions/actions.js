export const addProject = (project) => ({
  type: 'ADD_PROJECT',
  project
})