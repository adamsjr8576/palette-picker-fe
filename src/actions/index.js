export const addNewPalette = currentPalette => ({
  type: 'ADD_PALETTE',
  currentPalette
});

export const updatePaletteLocked = hexCode => ({
  type: 'UPDATE_PALETTE_LOCKED',
  hexCode
});

export const addInitialProjects = projects => ({
  type: 'ADD_INITIAL_PROJECTS',
  projects
});

export const addPalettes = palettes => ({
  type: 'ADD_PALETTE_TO_PROJECT',
  palettes
});

export const addProject = project => ({
  type: 'ADD_PROJECT',
  project
});

export const deletePaletteFromStore = palette => ({
  type: 'DELETE_PALETTE',
  palette
});

export const deleteProjectFromStore = project => ({
  type: 'DELETE_PROJECT',
  project
});
