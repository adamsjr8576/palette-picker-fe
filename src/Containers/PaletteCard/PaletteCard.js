import { connect } from 'react-redux';
import React from 'react';
import './PaletteCard.scss';
import images from '../../images/images';
import { updatePaletteLocked } from '../../actions';

export const PaletteCard = ({ locked, rightColor, leftColor, hexCode, updatePaletteLocked }) => {
  const image = locked? images.locked : images.unlocked;
  return (
    <article className="palette-article" style={{backgroundColor: `${hexCode}`}}>
      <div className="complimentary-colors-container">
        <div className="color-circle" style={{backgroundColor: `${rightColor}`}}></div>
        <div className="color-circle" style={{backgroundColor: `${leftColor}`}}></div>
      </div>
      <div className="palette-info-container">
        <section tabIndex="1"
          className="lock-button-section"
          role="update-palette=locked"
          onClick={() => updatePaletteLocked(hexCode)}
          onKeyDown={event => {
          if (event.key === "Enter") {
            updatePaletteLocked(hexCode);
          }
          }}>
          <img src={image} alt="lock icon" className="lock-button-image" />
        </section>
        <section className="hex-code-section">
          <p className="hex-p">{hexCode}</p>
        </section>
      </div>
    </article>
  );
};

export default PaletteCard;
