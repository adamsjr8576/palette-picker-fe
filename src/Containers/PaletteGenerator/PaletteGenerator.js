import { connect } from 'react-redux';
import React, { useEffect } from 'react';
import './PaletteGenerator.scss';
import { addNewPalette, updatePaletteLocked } from '../../actions';
import PaletteCard from '../PaletteCard/PaletteCard';

export const PaletteGenerator = ({ addNewPalette, currentPalette, updatePaletteLocked }) => {
  useEffect(() => {
    createPalette();
  }, []);

  let paletteCards;

  if (currentPalette.length) {
    paletteCards = currentPalette.map((color, index) => {
      return (
        <PaletteCard locked={color.locked} hexCode={color.color} key={index} id={color.color} updatePaletteLocked={updatePaletteLocked} />
      )
    });
  }

  const getRandomColor = () => {
    const randomColor = `#${Math.floor(Math.random()*16777215).toString(16)}`;
    return randomColor;
  }

  const createPalette = () => {
    let palette;
    if (currentPalette.length) {
      palette = currentPalette.map(palette => {
        if (!palette.locked) {
          const newColor = getRandomColor();
          return {locked: palette.locked, color: newColor}
        }
        return palette;
      });
    } else {
      palette = [
        {locked: false, color: getRandomColor()},
        {locked: false, color: getRandomColor()},
        {locked: false, color: getRandomColor()},
        {locked: false, color: getRandomColor()},
        {locked: false, color: getRandomColor()}]
    }
    addNewPalette(palette);
  }

  return (
    <section className="palette-section">
      <section className="palette-card-section">
        {paletteCards}
      </section>
      <button type="button" className="palette-button" onClick={() => createPalette()}>Create New Palette!</button>
    </section>
  );
};

export const mapDispatchToProps = dispatch => ({
  addNewPalette: currentPalette => dispatch( addNewPalette(currentPalette) ),
  updatePaletteLocked: hexCode => dispatch( updatePaletteLocked(hexCode) )
});

export const mapStateToProps = state => ({
  currentPalette: state.currentPalette
});

export default connect(mapStateToProps, mapDispatchToProps)(PaletteGenerator);
