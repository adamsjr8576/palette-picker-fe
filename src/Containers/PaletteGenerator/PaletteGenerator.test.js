import React from 'react';
import { shallow } from 'enzyme';
import { PaletteGenerator, mapDispatchToProps, mapStateToProps } from './PaletteGenerator';
import { PaletteCard } from '../PaletteCard/PaletteCard';
import { addPalette } from '../../actions/index';
import { render, fireEvent } from '@testing-library/react';

describe('PaletteGenerator', () => {

  let mockProps, setup;

  describe('PalleteGenerator', () => {

    test('should match the snapshot when currentPalette is an empty array', () => {
      mockProps = {
        addNewPalette: jest.fn(),
        currentPalette: []
      }

      setup = () => {
        const utils = render(<PaletteGenerator
          addNewPalette={mockProps.addNewPalette}
          currentPalette={mockProps.currentPalette}
        />)
        return { utils };
      }

      const { utils } = setup();
      expect(utils).toMatchSnapshot();
    });

    test('should match the snapshot when currentPalette is an array of colors', () => {
      mockProps = {
        addNewPalette: jest.fn(),
        currentPalette: [
          {locked: false, color: '#11111'},
          {locked: false, color: '#11111'},
          {locked: false, color: '#11111'},
          {locked: false, color: '#11111'},
          {locked: false, color: '#11111'}],
        updatePaletteLocked: jest.fn()
      }

      const mockMath = Object.create(global.Math);
      mockMath.floor = () => 11111;
      global.Math = mockMath;

      setup = () => {
        const utils = render(<PaletteGenerator
          addNewPalette={mockProps.addNewPalette}
          currentPalette={mockProps.currentPalette}
          updatePaletteLocked={mockProps.updatePaletteLocked}
        />);
        return { utils }
      }

      const { utils } = setup();
      expect(utils).toMatchSnapshot();
    });

    test('Should invoke createPalette on button click and when currentPalette is an empty array', () => {
      mockProps = {
        addNewPalette: jest.fn(),
        currentPalette: [],
        updatePaletteLocked: jest.fn()
      }

      const mockPaletteInitial = [
        {locked: false, color: '#444343'},
        {locked: false, color: '#6D6D6D'},
        {locked: false, color: '#9B9B9B'},
        {locked: false, color: '#C2C2C2'},
        {locked: false, color: '#DCDCDC'}
      ];

      setup = () => {
        const utils = render(<PaletteGenerator
          addNewPalette={mockProps.addNewPalette}
          currentPalette={mockProps.currentPalette}
          updatePaletteLocked={mockProps.updatePaletteLocked}
        />);
        const btn = utils.getByRole('create-palette')
        return {
          btn,
          ...utils
        };
      }

      const { btn } = setup();
      fireEvent.click(btn);
      expect(mockProps.addNewPalette).toHaveBeenCalledWith(mockPaletteInitial);
      expect(mockProps.addNewPalette).toHaveBeenCalledTimes(2);
    });

    test('Should invoke createPalette on button click and when currentPalette has colors it should reassign them', () => {
      mockProps = {
        addNewPalette: jest.fn(),
        currentPalette: [
          {locked: false, color: '#444343'},
          {locked: false, color: '#6D6D6D'},
          {locked: false, color: '#9B9B9B'},
          {locked: false, color: '#C2C2C2'},
          {locked: false, color: '#DCDCDC'}
        ],
        updatePaletteLocked: jest.fn()
      }

      const mockPalette = [
      { locked: false, color: '#2b67' },
      { locked: false, color: '#2b67' },
      { locked: false, color: '#2b67' },
      { locked: false, color: '#2b67' },
      { locked: false, color: '#2b67' }
    ];

      const mockMath = Object.create(global.Math);
      mockMath.floor = () => 11111;
      global.Math = mockMath;

      setup = () => {
        const utils = render(<PaletteGenerator
          addNewPalette={mockProps.addNewPalette}
          currentPalette={mockProps.currentPalette}
          updatePaletteLocked={mockProps.updatePaletteLocked}
        />);
        const btn = utils.getByRole('create-palette')
        return {
          btn,
          ...utils
        };
      }

      const { btn } = setup();
      fireEvent.click(btn);
      expect(mockProps.addNewPalette).toHaveBeenCalledWith(mockPalette);
      expect(mockProps.addNewPalette).toHaveBeenCalledTimes(2);
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with the currentPalette data', () => {
      const mockState = {
        currentPalette: [
          {locked: false, color: '#444343'},
          {locked: false, color: '#6D6D6D'},
          {locked: false, color: '#9B9B9B'},
          {locked: false, color: '#C2C2C2'},
          {locked: false, color: '#DCDCDC'}],
        projects: []
      };
      const expected = {
        currentPalette: [
          {locked: false, color: '#444343'},
          {locked: false, color: '#6D6D6D'},
          {locked: false, color: '#9B9B9B'},
          {locked: false, color: '#C2C2C2'},
          {locked: false, color: '#DCDCDC'}]
      };

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});
