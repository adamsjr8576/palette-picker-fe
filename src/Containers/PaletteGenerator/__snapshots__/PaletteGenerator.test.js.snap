// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteGenerator PalleteGenerator should match the snapshot when currentPalette is an array of colors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="palette-section"
      >
        <section
          class="palette-card-section"
        >
          <article
            class="palette-article"
          >
            <div
              class="palette-info-container"
            >
              <section
                class="lock-button-section"
                tabindex="1"
              >
                <img
                  alt="lock icon"
                  class="lock-button-image"
                  src="unlock.png"
                />
              </section>
              <section
                class="hex-code-section"
              >
                <p
                  class="hex-p"
                >
                  #11111
                </p>
              </section>
            </div>
          </article>
          <article
            class="palette-article"
          >
            <div
              class="palette-info-container"
            >
              <section
                class="lock-button-section"
                tabindex="1"
              >
                <img
                  alt="lock icon"
                  class="lock-button-image"
                  src="unlock.png"
                />
              </section>
              <section
                class="hex-code-section"
              >
                <p
                  class="hex-p"
                >
                  #11111
                </p>
              </section>
            </div>
          </article>
          <article
            class="palette-article"
          >
            <div
              class="palette-info-container"
            >
              <section
                class="lock-button-section"
                tabindex="1"
              >
                <img
                  alt="lock icon"
                  class="lock-button-image"
                  src="unlock.png"
                />
              </section>
              <section
                class="hex-code-section"
              >
                <p
                  class="hex-p"
                >
                  #11111
                </p>
              </section>
            </div>
          </article>
          <article
            class="palette-article"
          >
            <div
              class="palette-info-container"
            >
              <section
                class="lock-button-section"
                tabindex="1"
              >
                <img
                  alt="lock icon"
                  class="lock-button-image"
                  src="unlock.png"
                />
              </section>
              <section
                class="hex-code-section"
              >
                <p
                  class="hex-p"
                >
                  #11111
                </p>
              </section>
            </div>
          </article>
          <article
            class="palette-article"
          >
            <div
              class="palette-info-container"
            >
              <section
                class="lock-button-section"
                tabindex="1"
              >
                <img
                  alt="lock icon"
                  class="lock-button-image"
                  src="unlock.png"
                />
              </section>
              <section
                class="hex-code-section"
              >
                <p
                  class="hex-p"
                >
                  #11111
                </p>
              </section>
            </div>
          </article>
        </section>
        <button
          class="palette-button"
          role="create-palette"
          type="button"
        >
          Create New Palette!
        </button>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="palette-section"
    >
      <section
        class="palette-card-section"
      >
        <article
          class="palette-article"
        >
          <div
            class="palette-info-container"
          >
            <section
              class="lock-button-section"
              tabindex="1"
            >
              <img
                alt="lock icon"
                class="lock-button-image"
                src="unlock.png"
              />
            </section>
            <section
              class="hex-code-section"
            >
              <p
                class="hex-p"
              >
                #11111
              </p>
            </section>
          </div>
        </article>
        <article
          class="palette-article"
        >
          <div
            class="palette-info-container"
          >
            <section
              class="lock-button-section"
              tabindex="1"
            >
              <img
                alt="lock icon"
                class="lock-button-image"
                src="unlock.png"
              />
            </section>
            <section
              class="hex-code-section"
            >
              <p
                class="hex-p"
              >
                #11111
              </p>
            </section>
          </div>
        </article>
        <article
          class="palette-article"
        >
          <div
            class="palette-info-container"
          >
            <section
              class="lock-button-section"
              tabindex="1"
            >
              <img
                alt="lock icon"
                class="lock-button-image"
                src="unlock.png"
              />
            </section>
            <section
              class="hex-code-section"
            >
              <p
                class="hex-p"
              >
                #11111
              </p>
            </section>
          </div>
        </article>
        <article
          class="palette-article"
        >
          <div
            class="palette-info-container"
          >
            <section
              class="lock-button-section"
              tabindex="1"
            >
              <img
                alt="lock icon"
                class="lock-button-image"
                src="unlock.png"
              />
            </section>
            <section
              class="hex-code-section"
            >
              <p
                class="hex-p"
              >
                #11111
              </p>
            </section>
          </div>
        </article>
        <article
          class="palette-article"
        >
          <div
            class="palette-info-container"
          >
            <section
              class="lock-button-section"
              tabindex="1"
            >
              <img
                alt="lock icon"
                class="lock-button-image"
                src="unlock.png"
              />
            </section>
            <section
              class="hex-code-section"
            >
              <p
                class="hex-p"
              >
                #11111
              </p>
            </section>
          </div>
        </article>
      </section>
      <button
        class="palette-button"
        role="create-palette"
        type="button"
      >
        Create New Palette!
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PaletteGenerator PalleteGenerator should match the snapshot when currentPalette is an empty array 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="palette-section"
      >
        <section
          class="palette-card-section"
        />
        <button
          class="palette-button"
          role="create-palette"
          type="button"
        >
          Create New Palette!
        </button>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="palette-section"
    >
      <section
        class="palette-card-section"
      />
      <button
        class="palette-button"
        role="create-palette"
        type="button"
      >
        Create New Palette!
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
